packages/zip-it-and-ship-it/src/config.ts

```diff
@@ -3,32 +3,35 @@ import { basename, extname, dirname, join } from 'path'
 
 import isPathInside from 'is-path-inside'
 import mergeOptions from 'merge-options'
+import { z } from 'zod'
 
 import { FunctionSource } from './function.js'
-import type { NodeBundlerName } from './runtimes/node/bundlers/types.js'
-import type { ModuleFormat } from './runtimes/node/utils/module_format.js'
+import { nodeBundler } from './runtimes/node/bundlers/types.js'
+import { moduleFormat } from './runtimes/node/utils/module_format.js'
 import { minimatch } from './utils/matching.js'
 
-interface FunctionConfig {
-  externalNodeModules?: string[]
-  includedFiles?: string[]
-  includedFilesBasePath?: string
-  ignoredNodeModules?: string[]
-  nodeBundler?: NodeBundlerName
-  nodeSourcemap?: boolean
-  nodeVersion?: string
-  rustTargetDirectory?: string
-  schedule?: string
-  zipGo?: boolean
-  name?: string
-  generator?: string
-  timeout?: number
+export const functionConfig = z.object({
+  externalNodeModules: z.array(z.string()).optional().catch([]),
+  generator: z.string().optional().catch(undefined),
+  includedFiles: z.array(z.string()).optional().catch([]),
+  includedFilesBasePath: z.string().optional().catch(undefined),
+  ignoredNodeModules: z.array(z.string()).optional().catch([]),
+  name: z.string().optional().catch(undefined),
```

Comment:
there's a lot of duplication in this. does mit make sense to extract `z.string().optional()....` into a few constants, just so that there's fewer characters on the screen? It reads *very* dense right now.

Reply:
I see what you mean, but at the same time I don't know what would be the criteria for extracting something into a constant. We could have `optionalString = z.string()).optional()`, but is that really saving a lot of characters? We could add `.catch(undefined)` to that, but what do we do about the strings where we don't want to have a `catch`? I'm not super convinced this would help a lot, tbh.

Reply:
Are there any strings where we don't have a `catch`, though?

Maybe it'd already help to add a few empty lines in between the different parameters. That'd make it less of a uniform wall of text.